ScratchDjango/User/managers.py

```diff
@@ -0,0 +1,34 @@
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth.models import UserManager as DjangoUserManager
+from django.contrib.auth.password_validation import validate_password
+
+
+class UserManager(DjangoUserManager):
+    def _create_user(self, email: str, password: str | None, **extra_fields):
+        """
+        Create and save a user with the given email and password.
+        """
+        if not email:
+            raise ValueError("The given email must be set")
+        validate_password(password)
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.password = make_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email: str, password: str | None = None, **extra_fields):
+        extra_fields.setdefault("is_staff", False)
+        extra_fields.setdefault("is_superuser", False)
+        return self._create_user(email, password, **extra_fields)
+
+    def create_superuser(self, email: str, password: str | None = None, **extra_fields):
```

Comment:
Except of creating two functions, can we pass `is_staff` and `is_superuser` as variable and change the value according to user type

Reply:
The 2 fn are already in django User abstract model. We are only overriding those. This user manager was created by the cookiecutter template and I just copied it from there. What it does is that it makes the user register using email instead of default username. The create_superuser is the fn used when calling the python manage.py createsuperuser and it will throw error if I remove this overridden implementation of that create_superuser fn since it will not get the username to register the user because we have defined in User model to register using email.